src/OpenLoco/src/GameCommands/Road/RemoveRoad.cpp

```diff
@@ -0,0 +1,262 @@
+        if (roadElPiece == nullptr)
+        {
+            return 0;
+        }
+
+        currency32_t totalRemovalCost = 0;
+
+        const auto* roadObj = ObjectManager::get<RoadObject>(roadElPiece->roadObjectId());
+        {
+            const auto trackBaseCost = Economy::getInflationAdjustedCost(roadObj->sellCostFactor, roadObj->costIndex, 10);
+            const auto cost = (trackBaseCost * World::TrackData::getTrackMiscData(args.roadId).costFactor) / 256;
+            totalRemovalCost += cost;
+        }
+
+        // Check mod removal costs
+        for (auto i = 0U; i < 2; i++)
+        {
+            if (roadElPiece->hasMod(i))
+            {
+                const auto* trackExtraObj = ObjectManager::get<RoadExtraObject>(roadObj->mods[i]);
+                const auto trackExtraBaseCost = Economy::getInflationAdjustedCost(trackExtraObj->sellCostFactor, trackExtraObj->costIndex, 10);
+                const auto cost = (trackExtraBaseCost * World::TrackData::getTrackMiscData(args.roadId).costFactor) / 256;
+                totalRemovalCost += cost;
+            }
+        }
+
+        return totalRemovalCost;
+    }
+
+    // 0x004775A5
+    static uint32_t removeRoad(const RoadRemovalArgs& args, uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::Construction);
+        setPosition(args.pos + World::Pos3{ 16, 16, 0 });
+
+        // 0x0047762D
+        auto tile = World::TileManager::get(args.pos.x, args.pos.y);
+
+        if (auto* roadEl = getRoadElement(tile, args, args.sequenceIndex, flags); roadEl != nullptr)
+        {
+            const auto companyId = SceneManager::isEditorMode() ? CompanyId::neutral : getUpdatingCompanyId();
+            if (!sub_431E6A(companyId, reinterpret_cast<const World::TileElement*>(roadEl)))
+            {
+                return FAILURE;
+            }
+
+            /*
+            // This part was made redundant by a no-op write in Hooks.cpp:
+            // writeNop(0x004776DD, 6);
+            // TODO: turn this into a setting?
+            if (companyId != CompanyId::neutral && roadEl->hasUnk7_40() || roadEl->hasUnk7_80())
+            {
+                setErrorText(StringIds::empty);
+
+                auto nearest = TownManager::getClosestTownAndDensity(args.pos);
+                if (nearest.has_value())
+                {
+                    auto* town = TownManager::get(nearest->first);
+                    FormatArguments::common(town->name);
+                    setErrorText(StringIds::stringid_local_authority_wont_allow_removal_in_use);
+                    return FAILURE;
+                }
+            }
+            */
+        }
+
+        currency32_t totalRemovalCost = 0;
+
+        // 0x004776E3
+        if (auto* roadEl = getRoadElement(tile, args, args.sequenceIndex, flags); roadEl != nullptr)
+        {
+            auto* nextEl = roadEl->next();
+            auto* nextRoad = nextEl->as<World::RoadElement>();
+            if (nextRoad != nullptr)
```

Comment:
you are missing a check to see if roadEl is the first element on the tile and if it **is not** then get the previous element and if that is a roadElement then check if there is a station element. I think what all of this is saying is only remove the station element if there is only one road element here with a station element. Tbh the logic here is odd I don't quite understand what it is trying to do but it doesn't look like what we have here is correct either.